src/crappy/camera/opencv_camera_v4l2.py

```diff
@@ -61,67 +56,11 @@ def open(self, device_num: int = 0, **kwargs) -> None:
     self.log(logging.INFO, "Opening the image stream from the camera")
     self._cap = cv2.VideoCapture(device_num)
     self._device_num = device_num
-    fourcc = self._get_fourcc()
-
-    if system() == 'Linux':
-      self._formats = []
-
-      # Trying to run v4l2-ctl to get the available formats
-      command = ['v4l2-ctl', '-d', str(device_num), '--list-formats-ext']
-      try:
-        self.log(logging.INFO, f"Getting the available image formats with "
-                               f"command {command}")
-        check = run(command, capture_output=True, text=True)
-      except FileNotFoundError:
-        self.log(logging.WARNING, "The performance of the CameraOpencv class "
-                                  "could be improved if v4l-utils was "
-                                  "installed !")
-        check = None
-      check = check.stdout if check is not None else ''
-
-      # Splitting the returned string to isolate each encoding
-      if findall(r'\[\d+]', check):
-        check = split(r'\[\d+]', check)[1:]
-      elif findall(r'Pixel\sFormat', check):
-        check = split(r'Pixel\sFormat', check)[1:]
-      else:
-        check = []
-
-      if check:
-        for img_format in check:
-          # For each encoding, finding its name
-          name, *_ = search(r"'(\w+)'", img_format).groups()
-          sizes = findall(r'\d+x\d+', img_format)
-          fps_sections = split(r'\d+x\d+', img_format)[1:]
-
-          # For each name, finding the available sizes
-          for size, fps_section in zip(sizes, fps_sections):
-            fps_list = findall(r'\((\d+\.\d+)\sfps\)', fps_section)
-            for fps in fps_list:
-              self._formats.append(f'{name} {size} ({fps} fps)')
-
-      else:
-        # If v4l-utils is not installed, proposing two encodings without
-        # further detail
-        self._formats = [fourcc, 'MJPG']
-
-        # Still letting the user choose the size
-        self.add_scale_setting(name='width', lowest=1, highest=1920,
-                               getter=self._get_width, setter=self._set_width)
-        self.add_scale_setting(name='height', lowest=1, highest=1080,
-                               getter=self._get_height,
-                               setter=self._set_height)
-
-    else:
-      # On Windows the fourcc management is even messier than on Linux
-      self._formats = []
-
-      # Still letting the user choose the size
-      self.add_scale_setting(name='width', lowest=1, highest=1920,
-                             getter=self._get_width, setter=self._set_width)
-      self.add_scale_setting(name='height', lowest=1, highest=1080,
-                             getter=self._get_height, setter=self._set_height)
 
+    self._get_available_formats(device_num)
+    self._formats = [_format for _format in self._formats
```

Comment:
Indicate with a log message which formats have been discarded because they are not supported.